src/methods/is.js

```diff
@@ -108,6 +108,16 @@ export function IsCallable(realm: Realm, _func: Value): boolean {
   if (HasCompatibleType(func, FunctionValue)) return true;
   if (func.isSimpleObject()) return false;
 
+  if (func instanceof AbstractObjectValue && !func.values.isTop()) {
```

Comment:
Would it make sense to put this into AbstractObjectValue as a `mightNotBeCallable` method instead of throwing, or is this the only place where it's relevant?

Reply:
It would be useful if `typeof` could return an abstract boolean instead of throwing one way or another.

Moving it to AbstractObjectValue could work but there are a lot of internal methods. Some of them have the `Partial` suffix and allows abstract values to be passed or returned.

We could have an `IsCallablePartial` that returns an abstract boolean in this case.